mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java

```diff
@@ -78,12 +92,22 @@ public MesosSchedulerClient(
             .pipelineConfigurator(new HttpClientPipelineConfigurator<>())
             .build();
 
+        final UserAgent userAgent = new UserAgent(
+            UserAgent.userAgentEntryForGradleArtifact("rxnetty"),
+            UserAgent.userAgentEntryForMavenArtifact("org.apache.mesos.rx.java", "mesos-rxjava-core")
+        );
+
         createPost = (Send s) -> {
             final byte[] bytes = sendCodec.encode(s);
-            return just(HttpClientRequest.createPost("/api/v1/scheduler")
-                .withHeader("Content-Type", sendCodec.mediaType())
-                .withHeader("Accept", receiveCodec.mediaType())
-                .withContent(bytes)
+            final HttpClientRequest<ByteBuf> request =
+                HttpClientRequest.createPost("/api/v1/scheduler")
+                    .withHeader("Content-Type", sendCodec.mediaType())
+                    .withHeader("Accept", receiveCodec.mediaType())
+                    .withHeader("User-Agent", String.format("%s %s", name, userAgent.toString()));
```

Comment:
Should the `UserAgent` class be responsible for the full string? We could have `MesosSchedulerClient` take both the application name and an optional version string in its constructor, and use them to create the first entry for `UserAgent`. It would be more consistent than using `String.format` here.